src/cogent3/core/new_alignment.py

```diff
@@ -297,25 +310,62 @@ def __init__(
         *,
         seqs_data: SeqsData,
         moltype: new_moltype.MolType,
+        names: list[str] = None,
```

Comment:
Let's add `annotation_db` as an optional argument to the `SequenceCollection` constructor, and also add it to the `make_unaligned_seqs()` function. The default value should be configured the same way as it is in `alignment.py` (ie `DEFAULT_ANNOTATION_DB = BasicAnnotationDb`)